howso/attribute_maps.amlg

```diff
@@ -195,6 +195,31 @@
 		full_dependents_map
 	)
 
+	;Creates a mapping of shared deviations where the keys are each feature within a group of shared deviations, and the values are the feature names in
+	; which their deviations are stored under. For each group of shared deviations, they are stored under the first sorted feature in their group.
+	; e.g., a group of features with shared devations is [A, B, C], creates an mapping of A -> A, B -> A, C -> A
+	#!ComposeSharedDeviationsKeyMap
+	(let
+		(assoc shared_deviations_key_map (assoc))
+
+		(map
+			(lambda (let
+				;each group of shared deviations features is sorted
+				(assoc shared_features_list (sort (current_value 1)))
+				(map
+					(lambda (accum (assoc
+						shared_deviations_key_map
+							(append shared_deviations_key_map (associate (current_value 2) (first shared_features_list)))
+					)))
+					shared_features_list
+				)
+			))
+			shared_deviations_features
+		)
+
```

Comment:
is the (accum (assoc x (append x ...) ))
pattern here intentional? I'm struggling a bit to follow but accum-ing and append-ing to the same var at once seems strange